cloudevents/pydantic/event_attribute_types.py

```diff
@@ -0,0 +1,70 @@
+#  Copyright 2018-Present The CloudEvents Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import re
+
+import pydantic
+
+
+class AbsoluteURI(pydantic.ConstrainedStr):
+    """
+    Absolute URI as defined in RFC 3986 Section 4.3
+    https://datatracker.ietf.org/doc/html/rfc3986#section-4.3
+    """
+
+    regex = re.compile(
+        r"^([a-z0-9%_\-.~\[\]@!$&'()*+,;=:/?#]+:[a-z0-9%_\-.~\[\]@!$&'()*+,;=:/?#]*)?$",
+        re.UNICODE | re.IGNORECASE,
+    )
+
+
+class NonEmptyAbsoluteURI(AbsoluteURI):
+    """
+    Non-Empty Absolute URI as defined in RFC 3986 Section 4.3
+    https://datatracker.ietf.org/doc/html/rfc3986#section-4.3
+    """
+
+    min_length = 1
+
+
+class URIReference(pydantic.ConstrainedStr):
+    """
+    URI-reference as defined in RFC 3986 Section 4.1
+    https://datatracker.ietf.org/doc/html/rfc3986#section-4.1
+    """
+
+    regex = re.compile(
+        r"^[a-z0-9%_\-.~\[\]@!$&'()*+,;=:/?#]*$",
+        re.UNICODE | re.IGNORECASE,
+    )
+
+
+class NonEmptyURIReference(URIReference):
+    """
+    Non-Empty URI-reference as defined in RFC 3986 Section 4.1
+    https://datatracker.ietf.org/doc/html/rfc3986#section-4.1
+    """
+
+    min_length = 1
```

Comment:
Pydantic has built-in URI types, it's better to use one of that URIs: https://pydantic-docs.helpmanual.io/usage/types/#urls

Reply:
These types support only Urls and not URIs for example it fails on uuid urn
this one for example `urn:uuid:3703e79d-c635-4a50-bec1-e2b778e1de05`